pkg/compose/hash.go

```diff
@@ -40,3 +41,32 @@ func ServiceHash(o types.ServiceConfig) (string, error) {
 	}
 	return digest.SHA256.FromBytes(bytes).Encoded(), nil
 }
+
+func DeepCopy(src *types.VolumeConfig) (types.VolumeConfig, error) {
+	var dst types.VolumeConfig
+	data, err := json.Marshal(src)
+	if err != nil {
+		return types.VolumeConfig{}, err
+	}
+	err = json.Unmarshal(data, &dst)
+	return dst, err
+}
+
+// From a top-level Volume Configuration, creates a unique hash ignoring
+// External
+func VolumeHash(v types.VolumeConfig) (string, error) {
+	o, err := DeepCopy(&v)
+	if err != nil {
+		return "", err
+	}
+
+	if o.Driver == "" { // (TODO: jhrotko) This probably should be fixed in compose-go
+		o.Driver = "local"
+	}
+	delete(o.Labels, api.ConfigHashLabel)
```

Comment:
also need to ignore a few more labels, like ComposeVersionLabel (don't wan't to recreate all volumes after you updated compose, do you?)
See https://github.com/compose-spec/compose-go/pull/708 and https://github.com/docker/compose/pull/12267 for example usage